File: JovenVision.Domain/Entities/Group.cs
namespace JovenVision.Domain.Entities
{
    public class Group : EntityBase
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public int Capacity { get; set; }
        public string Status { get; set; } = "ACTIVE";
        public ICollection<GroupMember> GroupMembers { get; set; } = new List<GroupMember>();
    }
}

File: JovenVision.Web/src/types/group.ts
export type GroupStatus = 'ACTIVE' | 'INACTIVE';

export interface Group {
  id: number;
  name: string;
  description?: string;
  capacity: number;
  status: GroupStatus;
  createdAt: string;
  updatedAt?: string;
  deletedAt?: string;
}

export interface GroupPayload {
  name: string;
  description?: string;
  capacity: number;
  status: GroupStatus;
}

export interface GroupFilters {
  search?: string;
  status?: string;
  page?: number;
  pageSize?: number;
}

export interface GroupListResponse {
  data: Group[];
  totalCount: number;
  page: number;
  pageSize: number;
}
